lib/compress/zstd_compress.c

```diff
@@ -1594,60 +1669,88 @@ ZSTD_adjustCParams_internal(ZSTD_compressionParameters cPar,
 ZSTD_adjustCParams(ZSTD_compressionParameters cPar,
                    unsigned long long srcSize,
                    size_t dictSize)
 {
-    cPar = ZSTD_clampCParams(cPar);   /* resulting cPar is necessarily valid (all parameters within range) */
+    ZSTD_CCtx_params params;
+    ZSTD_CCtxParams_init(&params, 0);
+    params.cParams = cPar;
+    ZSTD_clampCParams(&params);   /* resulting cPar is necessarily valid (all parameters within range) */
     if (srcSize == 0) srcSize = ZSTD_CONTENTSIZE_UNKNOWN;
-    return ZSTD_adjustCParams_internal(cPar, srcSize, dictSize, ZSTD_cpm_unknown, ZSTD_ps_auto);
-}
-
-static ZSTD_compressionParameters ZSTD_getCParams_internal(int compressionLevel, unsigned long long srcSizeHint, size_t dictSize, ZSTD_CParamMode_e mode);
-static ZSTD_parameters ZSTD_getParams_internal(int compressionLevel, unsigned long long srcSizeHint, size_t dictSize, ZSTD_CParamMode_e mode);
-
-static void ZSTD_overrideCParams(
-              ZSTD_compressionParameters* cParams,
-        const ZSTD_compressionParameters* overrides)
-{
-    if (overrides->windowLog)    cParams->windowLog    = overrides->windowLog;
-    if (overrides->hashLog)      cParams->hashLog      = overrides->hashLog;
-    if (overrides->chainLog)     cParams->chainLog     = overrides->chainLog;
-    if (overrides->searchLog)    cParams->searchLog    = overrides->searchLog;
-    if (overrides->minMatch)     cParams->minMatch     = overrides->minMatch;
-    if (overrides->targetLength) cParams->targetLength = overrides->targetLength;
-    if (overrides->strategy)     cParams->strategy     = overrides->strategy;
-}
-
-ZSTD_compressionParameters ZSTD_getCParamsFromCCtxParams(
-        const ZSTD_CCtx_params* CCtxParams, U64 srcSizeHint, size_t dictSize, ZSTD_CParamMode_e mode)
-{
-    ZSTD_compressionParameters cParams;
-    if (srcSizeHint == ZSTD_CONTENTSIZE_UNKNOWN && CCtxParams->srcSizeHint > 0) {
-        assert(CCtxParams->srcSizeHint>=0);
-        srcSizeHint = (U64)CCtxParams->srcSizeHint;
+    ZSTD_adjustCParams_internal(&params, srcSize, dictSize, ZSTD_cpm_unknown, ZSTD_ps_auto);
+    return params.cParams;
+}
+
+static ZSTD_CCtx_params ZSTD_getCCtxParams_internal(
+        int compressionLevel,
+        unsigned long long srcSizeHint,
+        size_t dictSize,
+        ZSTD_CParamMode_e mode);
+
+/* Prefer ZSTD_getCCtxParams_internal() unless you're sure you *only* need
+ * the CParams actually stored in the ZSTD_compressionParameters struct.
+ * (i.e., not the window size!) */
+static ZSTD_compressionParameters ZSTD_getCParams_internal(
+        int compressionLevel,
+        unsigned long long srcSizeHint,
+        size_t dictSize,
+        ZSTD_CParamMode_e mode);
+static ZSTD_parameters ZSTD_getParams_internal(
+        int compressionLevel,
+        unsigned long long srcSizeHint,
+        size_t dictSize, ZSTD_CParamMode_e
+        mode);
+
+/* fills unset fields in `dst` with values from `defaults` */
+static void ZSTD_fillCParams(
```

Comment:
The naming feels weird.
What is being filled is a `ZSTD_CCtx_params*` object,
though admittedly on a reduced internal scope,
but is that actually a good idea to keep the same scope as it was before ?

It looks to me that this operation is about setting explicitly all parameters that are not yet set,
using a list provided in `defaults`.